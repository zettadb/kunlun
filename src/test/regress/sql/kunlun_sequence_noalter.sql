create sequence seq2;
create table tseq2(a serial primary key, b smallserial, c bigserial, d int auto_increment, e int generated by default as identity, f int default nextval('seq2')) partition by hash(a);
create table tseq21 partition of tseq2 for values with (modulus 4, remainder 0);
create table tseq22 partition of tseq2 for values with (modulus 4, remainder 1);
create table tseq23 partition of tseq2 for values with (modulus 4, remainder 2);
create table tseq24 partition of tseq2 for values with (modulus 4, remainder 3);
insert into tseq2(e, f) values(default, default),(34, default),(default, default),(45, default);
select*from tseq2;
insert into tseq2(e) values(11),(default),(33),(default);
select*from tseq2;
insert into tseq2(f) values(111),(default),(333),(default);
select*from tseq2;



-- oracle sequence grammar
create sequence seq34 nomaxvalue nominvalue nocache nocycle noorder starts with 34;
create sequence seq35 nomaxvalue nominvalue cache 1 no cycle order starts with 35;
select seq34.nextval;
select seq35.currval;
select seq35.nextval;
select seq35.currval;
drop sequence seq35;
drop sequence seq34;

-- seq value fetch
create table t6(a int auto_increment primary key, b serial, c int generated by default as identity,d int);
insert into t6(d) values(11),(22),(33);
insert into t6(d) values(11),(22),(33);
insert into t6(d) values(11),(22),(33);
insert into t6(d) values(11),(22),(33);
select*from t6;
select currval('t6_a_seq');
select currval('t6_b_seq');
select nextval('t6_b_seq');
select currval('t6_c_seq');



create sequence seq7;
create sequence seq6;

drop table if exists t7;
create table t7(a int auto_increment primary key, b serial, c int generated by default as identity,d int default nextval('seq7'), e int);
insert into t7 (e) values(11),(22),(33) returning *;
insert into t7 (e) values(11),(22),(33) returning *;
insert into t7 (e) values(11),(22),(33) returning *;
insert into t7 (e) values(11),(22),(33) returning *;
select*from t7;
select currval('t7_a_seq');
select currval('t7_b_seq');
select currval('t7_c_seq');
select currval('seq7');
select seq7.nextval, seq7.currval, seq7.nextval, seq7.currval;
select seq6.nextval, seq7.currval, seq7.nextval, seq6.currval;

insert into t6(c,d) values(seq6.nextval, seq7.nextval),(seq6.nextval, seq7.nextval),(seq6.currval, seq7.currval);
insert into t6(c,d) values(seq6.nextval, seq7.nextval),(seq6.nextval, seq7.nextval),(seq6.currval, seq7.currval);
insert into t6(c,d) values(seq6.nextval, seq7.nextval),(seq6.nextval, seq7.nextval),(seq6.currval, seq7.currval);
insert into t6(c,d) values(seq6.currval, seq7.currval),(seq6.currval, seq7.currval),(seq6.nextval, seq7.nextval);
select*from t6;



create schema scm1;
alter sequence seq7 set schema scm1;
-- alter sequence scm1.seq7 rename to scm1.seq77;
-- above doesn't work, new name can't have schema name prepended.
-- below stmt rename a seq but doesn't move it to public schema, seq77 is still in scm1.
alter sequence scm1.seq7 rename to seq77;

select scm1.seq77.nextval, scm1.seq77.currval;
select lastval();
drop schema scm1;
drop schema scm1 cascade;

drop table t6;
drop table t7;
drop table tseq2;
drop sequence seq2;
drop sequence seq7;
drop sequence seq6;
