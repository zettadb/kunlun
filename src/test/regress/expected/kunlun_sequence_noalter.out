create sequence seq2;
CREATE SEQUENCE
create table tseq2(a serial primary key, b smallserial, c bigserial, d int auto_increment, e int generated by default as identity, f int default nextval('seq2')) partition by hash(a);
CREATE TABLE
create table tseq21 partition of tseq2 for values with (modulus 4, remainder 0);
CREATE TABLE
create table tseq22 partition of tseq2 for values with (modulus 4, remainder 1);
CREATE TABLE
create table tseq23 partition of tseq2 for values with (modulus 4, remainder 2);
CREATE TABLE
create table tseq24 partition of tseq2 for values with (modulus 4, remainder 3);
CREATE TABLE
insert into tseq2(e, f) values(default, default),(34, default),(default, default),(45, default);
INSERT 0 4
select*from tseq2;
 a | b | c | d | e  | f 
---+---+---+---+----+---
 1 | 1 | 1 | 1 |  1 | 1
 3 | 3 | 3 | 3 |  2 | 3
 2 | 2 | 2 | 2 | 34 | 2
 4 | 4 | 4 | 4 | 45 | 4
(4 rows)

insert into tseq2(e) values(11),(default),(33),(default);
INSERT 0 4
select*from tseq2;
 a | b | c | d | e  | f 
---+---+---+---+----+---
 1 | 1 | 1 | 1 |  1 | 1
 3 | 3 | 3 | 3 |  2 | 3
 5 | 5 | 5 | 5 | 11 | 5
 8 | 8 | 8 | 8 |  4 | 8
 2 | 2 | 2 | 2 | 34 | 2
 4 | 4 | 4 | 4 | 45 | 4
 6 | 6 | 6 | 6 |  3 | 6
 7 | 7 | 7 | 7 | 33 | 7
(8 rows)

insert into tseq2(f) values(111),(default),(333),(default);
INSERT 0 4
select*from tseq2;
 a  | b  | c  | d  | e  |  f  
----+----+----+----+----+-----
  1 |  1 |  1 |  1 |  1 |   1
 12 | 12 | 12 | 12 |  8 |  10
  3 |  3 |  3 |  3 |  2 |   3
  5 |  5 |  5 |  5 | 11 |   5
  8 |  8 |  8 |  8 |  4 |   8
  9 |  9 |  9 |  9 |  5 | 111
 11 | 11 | 11 | 11 |  7 | 333
  2 |  2 |  2 |  2 | 34 |   2
  4 |  4 |  4 |  4 | 45 |   4
  6 |  6 |  6 |  6 |  3 |   6
  7 |  7 |  7 |  7 | 33 |   7
 10 | 10 | 10 | 10 |  6 |   9
(12 rows)

-- oracle sequence grammar
create sequence seq34 nomaxvalue nominvalue nocache nocycle noorder starts with 34;
CREATE SEQUENCE
create sequence seq35 nomaxvalue nominvalue cache 1 no cycle order starts with 35;
CREATE SEQUENCE
select seq34.nextval;
 nextval 
---------
      34
(1 row)

select seq35.currval;
psql:sql/kunlun_sequence_noalter.sql:20: ERROR:  currval of sequence "seq35" is not yet defined in this session
select seq35.nextval;
 nextval 
---------
      35
(1 row)

select seq35.currval;
 currval 
---------
      35
(1 row)

drop sequence seq35;
DROP SEQUENCE
drop sequence seq34;
DROP SEQUENCE
-- seq value fetch
create table t6(a int auto_increment primary key, b serial, c int generated by default as identity,d int);
CREATE TABLE
insert into t6(d) values(11),(22),(33);
INSERT 0 3
insert into t6(d) values(11),(22),(33);
INSERT 0 3
insert into t6(d) values(11),(22),(33);
INSERT 0 3
insert into t6(d) values(11),(22),(33);
INSERT 0 3
select*from t6;
 a  | b  | c  | d  
----+----+----+----
  1 |  1 |  1 | 11
  2 |  2 |  2 | 22
  3 |  3 |  3 | 33
  4 |  4 |  4 | 11
  5 |  5 |  5 | 22
  6 |  6 |  6 | 33
  7 |  7 |  7 | 11
  8 |  8 |  8 | 22
  9 |  9 |  9 | 33
 10 | 10 | 10 | 11
 11 | 11 | 11 | 22
 12 | 12 | 12 | 33
(12 rows)

select currval('t6_a_seq');
 currval 
---------
      12
(1 row)

select currval('t6_b_seq');
 currval 
---------
      12
(1 row)

select nextval('t6_b_seq');
 nextval 
---------
      13
(1 row)

select currval('t6_c_seq');
 currval 
---------
      12
(1 row)

create sequence seq7;
CREATE SEQUENCE
create sequence seq6;
CREATE SEQUENCE
drop table if exists t7;
DROP TABLE
create table t7(a int auto_increment primary key, b serial, c int generated by default as identity,d int default nextval('seq7'), e int);
CREATE TABLE
insert into t7 (e) values(11),(22),(33) returning *;
 a | b | c | d | e  
---+---+---+---+----
 1 | 1 | 1 | 1 | 11
 2 | 2 | 2 | 2 | 22
 3 | 3 | 3 | 3 | 33
(3 rows)

INSERT 0 3
insert into t7 (e) values(11),(22),(33) returning *;
 a | b | c | d | e  
---+---+---+---+----
 4 | 4 | 4 | 4 | 11
 5 | 5 | 5 | 5 | 22
 6 | 6 | 6 | 6 | 33
(3 rows)

INSERT 0 3
insert into t7 (e) values(11),(22),(33) returning *;
 a | b | c | d | e  
---+---+---+---+----
 7 | 7 | 7 | 7 | 11
 8 | 8 | 8 | 8 | 22
 9 | 9 | 9 | 9 | 33
(3 rows)

INSERT 0 3
insert into t7 (e) values(11),(22),(33) returning *;
 a  | b  | c  | d  | e  
----+----+----+----+----
 10 | 10 | 10 | 10 | 11
 11 | 11 | 11 | 11 | 22
 12 | 12 | 12 | 12 | 33
(3 rows)

INSERT 0 3
select*from t7;
 a  | b  | c  | d  | e  
----+----+----+----+----
  1 |  1 |  1 |  1 | 11
  2 |  2 |  2 |  2 | 22
  3 |  3 |  3 |  3 | 33
  4 |  4 |  4 |  4 | 11
  5 |  5 |  5 |  5 | 22
  6 |  6 |  6 |  6 | 33
  7 |  7 |  7 |  7 | 11
  8 |  8 |  8 |  8 | 22
  9 |  9 |  9 |  9 | 33
 10 | 10 | 10 | 10 | 11
 11 | 11 | 11 | 11 | 22
 12 | 12 | 12 | 12 | 33
(12 rows)

select currval('t7_a_seq');
 currval 
---------
      12
(1 row)

select currval('t7_b_seq');
 currval 
---------
      12
(1 row)

select currval('t7_c_seq');
 currval 
---------
      12
(1 row)

select currval('seq7');
 currval 
---------
      12
(1 row)

select seq7.nextval, seq7.currval, seq7.nextval, seq7.currval;
 nextval | currval | nextval | currval 
---------+---------+---------+---------
      13 |      13 |      14 |      14
(1 row)

select seq6.nextval, seq7.currval, seq7.nextval, seq6.currval;
 nextval | currval | nextval | currval 
---------+---------+---------+---------
       1 |      14 |      15 |       1
(1 row)

insert into t6(c,d) values(seq6.nextval, seq7.nextval),(seq6.nextval, seq7.nextval),(seq6.currval, seq7.currval);
INSERT 0 3
insert into t6(c,d) values(seq6.nextval, seq7.nextval),(seq6.nextval, seq7.nextval),(seq6.currval, seq7.currval);
INSERT 0 3
insert into t6(c,d) values(seq6.nextval, seq7.nextval),(seq6.nextval, seq7.nextval),(seq6.currval, seq7.currval);
INSERT 0 3
insert into t6(c,d) values(seq6.currval, seq7.currval),(seq6.currval, seq7.currval),(seq6.nextval, seq7.nextval);
INSERT 0 3
select*from t6;
 a  | b  | c  | d  
----+----+----+----
  1 |  1 |  1 | 11
  2 |  2 |  2 | 22
  3 |  3 |  3 | 33
  4 |  4 |  4 | 11
  5 |  5 |  5 | 22
  6 |  6 |  6 | 33
  7 |  7 |  7 | 11
  8 |  8 |  8 | 22
  9 |  9 |  9 | 33
 10 | 10 | 10 | 11
 11 | 11 | 11 | 22
 12 | 12 | 12 | 33
 13 | 14 |  2 | 16
 14 | 15 |  3 | 17
 15 | 16 |  3 | 17
 16 | 17 |  4 | 18
 17 | 18 |  5 | 19
 18 | 19 |  5 | 19
 19 | 20 |  6 | 20
 20 | 21 |  7 | 21
 21 | 22 |  7 | 21
 22 | 23 |  7 | 21
 23 | 24 |  7 | 21
 24 | 25 |  8 | 22
(24 rows)

create schema scm1;
CREATE SCHEMA
alter sequence seq7 set schema scm1;
ALTER SEQUENCE
-- alter sequence scm1.seq7 rename to scm1.seq77;
-- above doesn't work, new name can't have schema name prepended.
-- below stmt rename a seq but doesn't move it to public schema, seq77 is still in scm1.
alter sequence scm1.seq7 rename to seq77;
ALTER SEQUENCE
select scm1.seq77.nextval, scm1.seq77.currval;
 nextval | currval 
---------+---------
      23 |      23
(1 row)

select lastval();
 lastval 
---------
      23
(1 row)

drop schema scm1;
psql:sql/kunlun_sequence_noalter.sql:74: ERROR:  cannot drop schema scm1 because other objects depend on it
DETAIL:  sequence scm1.seq77 depends on schema scm1
default value for column d of table t7 depends on sequence scm1.seq77
HINT:  Use DROP ... CASCADE to drop the dependent objects too.
drop schema scm1 cascade;
psql:sql/kunlun_sequence_noalter.sql:75: NOTICE:  drop cascades to 2 other objects
DETAIL:  drop cascades to sequence scm1.seq77
drop cascades to default value for column d of table t7
DROP SCHEMA
drop table t6;
DROP TABLE
drop table t7;
DROP TABLE
drop table tseq2;
DROP TABLE
drop sequence seq2;
DROP SEQUENCE
drop sequence seq7;
psql:sql/kunlun_sequence_noalter.sql:81: ERROR:  sequence "seq7" does not exist
drop sequence seq6;
DROP SEQUENCE
